src/compiler/path.ts

```diff
@@ -155,18 +156,19 @@ namespace ts {
                 const scheme = path.slice(0, schemeEnd);
                 const authority = path.slice(authorityStart, authorityEnd);
                 if (scheme === "file" && (authority === "" || authority === "localhost") &&
+                    authorityEnd + 1 < path.length &&
                     isVolumeCharacter(path.charCodeAt(authorityEnd + 1))) {
                     const volumeSeparatorEnd = getFileUrlVolumeSeparatorEnd(path, authorityEnd + 2);
                     if (volumeSeparatorEnd !== -1) {
-                        if (path.charCodeAt(volumeSeparatorEnd) === CharacterCodes.slash) {
-                            // URL: "file:///c:/", "file://localhost/c:/", "file:///c%3a/", "file://localhost/c%3a/"
-                            return ~(volumeSeparatorEnd + 1);
-                        }
                         if (volumeSeparatorEnd === path.length) {
                             // URL: "file:///c:", "file://localhost/c:", "file:///c$3a", "file://localhost/c%3a"
                             // but not "file:///c:d" or "file:///c%3ad"
                             return ~volumeSeparatorEnd;
                         }
+                        if (path.charCodeAt(volumeSeparatorEnd) === CharacterCodes.slash) {

```

Comment:
Why'd this move lower?

Reply:
It seemed silly to do the length check after accessing the character that position.

Reply:
Gotcha. Looking at it more closely, I don't even know if `volumeSeparatorEnd === path.length` is even a valid condition, but I'm not familiar enough with this code.